tests/Unit/FileSelectionTest.php

```diff
@@ -34,9 +33,13 @@ public function it_can_select_all_the_files_in_a_directory_and_subdirectories()
                 'file1.txt',
                 'file2.txt',
                 'file3.txt',
-            ]),
-            iterator_to_array($fileSelection->selectedFiles())
-        );
+            ]);
+        $selectedFiles = iterator_to_array($fileSelection->selectedFiles());
+
+        sort($testFiles);
+        sort($selectedFiles);
+
+        $this->assertSame($testFiles, $selectedFiles);
```

Comment:
Instead of doing this check over and over again could you add a new protected method `assertSameArray` where in you sort both arrays and perform `assertSame`?